### ISSUE TITLE
[Bug]: dn start failed from commit: 47684bd to commit: 3a644a9

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#11311]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

panic log:
panic: vector checkpoint_meta_block_location not found
2023-08-19 01:12:01.039	
	/go/src/github.com/matrixorigin/matrixone/pkg/common/stopper/stopper.go:254 +0xdc
2023-08-19 01:12:01.039	
created by github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask
2023-08-19 01:12:01.039	
	/go/src/github.com/matrixorigin/matrixone/pkg/common/stopper/stopper.go:259 +0x6c
2023-08-19 01:12:01.039	
github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask.func1()
2023-08-19 01:12:01.039	
	/go/src/github.com/matrixorigin/matrixone/pkg/dnservice/store.go:286 +0x144
2023-08-19 01:12:01.039	
github.com/matrixorigin/matrixone/pkg/dnservice.(*store).createReplica.func1({0x2c0de58?, 0x4001551140?})
2023-08-19 01:12:01.039	
	/go/src/github.com/matrixorigin/matrixone/pkg/dnservice/factory.go:73 +0x220
2023-08-19 01:12:01.039	
github.com/matrixorigin/matrixone/pkg/dnservice.(*store).createTxnStorage(0x4001a38340, {0x2c0de58, 0x4001542960}, {{0x2, 0x1, {}, {0x0, 0x0, 0x0}, 0x0}, ...})
2023-08-19 01:12:01.039	
	/go/src/github.com/matrixorigin/matrixone/pkg/dnservice/factory.go:151 +0x1f4
2023-08-19 01:12:01.039	
github.com/matrixorigin/matrixone/pkg/dnservice.(*store).newTAEStorage(0x4001a38340, {0x2c0de58, 0x4001542960}, {{0x2, 0x1, {}, {0x0, 0x0, 0x0}, 0x0}, ...}, ...)
2023-08-19 01:12:01.039	
	/go/src/github.com/matrixorigin/matrixone/pkg/txn/storage/tae/storage.go:69 +0x1cc
2023-08-19 01:12:01.039	
github.com/matrixorigin/matrixone/pkg/txn/storage/tae.NewTAEStorage({0x2c0de58?, 0x4001542960}, {0x4001a080f0, 0x4c}, {{0x2, 0x1, {}, {0x0, 0x0, 0x0}, ...}, ...}, ...)
2023-08-19 01:12:01.039	
	/go/src/github.com/matrixorigin/matrixone/pkg/vm/engine/tae/rpc/handle.go:96 +0x40
2023-08-19 01:12:01.039	
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/rpc.NewTAEHandle({0x2c0de58?, 0x4001542960?}, {0x4001a080f0?, 0x0?}, 0x40014e7b28?)
2023-08-19 01:12:01.039	
	/go/src/github.com/matrixorigin/matrixone/pkg/vm/engine/tae/rpc/handle.go:1097 +0x130
2023-08-19 01:12:01.039	
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/rpc.openTAE({0x2c0de58, 0x4001542960}, {0x4001a080f0, 0x4c}, 0x100?)
2023-08-19 01:12:01.039	
	/go/src/github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db/open.go:143 +0xd9c
2023-08-19 01:12:01.039	
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db.Open({0x2c0de58, 0x4001542960}, {0x400005c500, 0x50}, 0x40014b8d00?)
2023-08-19 01:12:01.039	
	/go/src/github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db/checkpoint/replay.go:251 +0x147c
2023-08-19 01:12:01.039	
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db/checkpoint.(*runner).Replay(0x4000ef6600, {0x2c09170, 0x400196e000})
2023-08-19 01:12:01.039	
	/go/src/github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db/checkpoint/replay.go:167 +0x44
2023-08-19 01:12:01.039	
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db/checkpoint.(*runner).Replay.func3()
2023-08-19 01:12:01.039	
	/go/src/github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db/checkpoint/replay.go:161 +0x3c
2023-08-19 01:12:01.039	
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db/checkpoint.(*runner).Replay.func2.1()
2023-08-19 01:12:01.039	
	/go/src/github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logtail/utils.go:1966 +0x1e8
2023-08-19 01:12:01.039	
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logtail.(*CheckpointData).CloseWhenLoadFromCache(0x4001972ee0, 0x4)
2023-08-19 01:12:01.039	
	/go/src/github.com/matrixorigin/matrixone/pkg/vm/engine/tae/containers/batch.go:100 +0xd0
2023-08-19 01:12:01.039	
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/containers.(*Batch).GetVectorByName(0x40076a8730, {0x27df539, 0x1e})

job link:
https://github.com/matrixorigin/mo-nightly-regression/actions/runs/5905197193/job/16018929741

<img width="643" alt="image" src="https://github.com/matrixorigin/matrixone/assets/47846308/96bab2e1-79c3-4ab1-97a8-fe68fd0e0eba">

log from loki:

http://a58ec25c4c81b42b4bc11f9a4eb1f54e-1422586515.us-west-2.elb.amazonaws.com/explore?orgId=1&left=%7B%22datasource%22:%22b08c0b34-9dfc-47c9-ac7b-b1dd15d38191%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22mo-cpt%5C%22,%20pod%3D%5C%22cpt-dn-0%5C%22%7D%20%7C%3D%20%60%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22b08c0b34-9dfc-47c9-ac7b-b1dd15d38191%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%221692335587108%22,%22to%22:%221692378787108%22%7D%7D

**Comment by jiangxinmeng1** [comment]
fixed by #11316

**Comment by aressu1985** [comment]
fixed
